crates/swc_ecma_utils/src/lib.rs

```diff
@@ -379,29 +374,55 @@ pub trait StmtExt {
                 .collect(),
         })
     }
+
+    /// stmts contain top level return/break/continue/throw
+    fn terminates(&self) -> bool;
 }
 
 impl StmtExt for Stmt {
-    #[inline]
-    fn into_stmt(self) -> Stmt {
-        self
+    fn extract_var_ids(&self) -> Vec<Ident> {
+        extract_var_ids(self)
     }
 
-    #[inline]
-    fn as_stmt(&self) -> &Stmt {
-        self
+    fn terminates(&self) -> bool {
+        match self {
+            Stmt::Break(_) | Stmt::Continue(_) | Stmt::Throw(_) | Stmt::Return(_) => return true,
+            Stmt::Block(block) if block.stmts.terminates() => return true,
+            Stmt::If(IfStmt {
+                cons,
+                alt: Some(alt),
+                ..
+            }) if cons.terminates() && alt.terminates() => return true,
```

Comment:
It would be better to remove if guard and reutrn `cons.terminates() && alt.terminates()`